pkg/cmd/release/create/http.go

```diff
@@ -174,6 +178,20 @@ func createRelease(httpClient *http.Client, repo ghrepo.Interface, params map[st
 	}
 	defer resp.Body.Close()
 
+	// Releases with workflow files can be created without the workflow scope,
+	// if the same file (with both the same path and contents) exists
+	// on another branch in the repo. Otherwise, the workflow scope is required.
+	oAuthTokenMissingWorkflowScope := isOAuthToken(resp) && !oAuthTokenHasWorkflowScope(resp)
+	if resp.StatusCode == 404 && oAuthTokenMissingWorkflowScope {
+		normalizedHostname := ghauth.NormalizeHostname(resp.Request.URL.Hostname())
+		errMissingRequiredWorkflowScope := errors.New(heredoc.Docf(`
+				HTTP 404: Failed to create release, "workflow" scope may be required
+				To request it, run gh auth refresh -h %[1]s -s workflow
+			`, normalizedHostname))
+
+		return nil, errMissingRequiredWorkflowScope
+	}
+
```

Comment:
I'm not a fan of how this is written and am looking for feedback on ways to improve it.

Reply:
Honestly, I'm torn here and left https://github.com/cli/cli/issues/9773#issuecomment-2432134253 to explain.  I think this might be an opportunity to try improving the REST API endpoint behavior rather than trying to work around it in `gh`.

Beyond that, here are other places in code where we handle similar situations:

https://github.com/cli/cli/blob/e390874fef5eb5545db2c4ff0861cb1c3e83dd6f/pkg/cmd/auth/setupgit/setupgit.go#L87-L100

https://github.com/cli/cli/blob/e390874fef5eb5545db2c4ff0861cb1c3e83dd6f/pkg/cmd/gpg-key/add/add.go#L84-L105

https://github.com/cli/cli/blob/e390874fef5eb5545db2c4ff0861cb1c3e83dd6f/pkg/cmd/project/shared/queries/queries.go#L1458-L1466